File: cpp_merge/test_hybrid/src/logger.cpp
#include "logger.hpp"

Logger::Logger(string prefix) {
    this->prefix = prefix;
}

void Logger::log(string level, string message)
{
    auto timestamp = ((std::make_shared<Date>()))->toISOString();
    console->log(std:("[") + timestamp + std:("] [") + this->prefix + std:("] [") + level + std:("] ") + message + string_empty);
}

void Logger::info(string message)
{
    this->log(std:("INFO"), message);
}

void Logger::warn(string message)
{
    this->log(std:("WARN"), message);
}

void Logger::error(string message)
{
    this->log(std:("ERROR"), message);
}

void Logger::debug(string message)
{
    this->log(std:("DEBUG"), message);
}

std::shared_ptr<Logger> createLogger(string prefix)
{
    return std::make_shared<Logger>(prefix);
};

File: classified/packages/plugin-stagehand/stagehand-server/src/logger.ts
export class Logger {
  private prefix = '[StagehandServer]';

  info(message: string, ...args: any[]) {
    console.log(`${this.prefix} INFO:`, message, ...args);
  }

  debug(message: string, ...args: any[]) {
    if (process.env.DEBUG) {
      console.log(`${this.prefix} DEBUG:`, message, ...args);
    }
  }

  error(message: string, error?: any) {
    console.error(`${this.prefix} ERROR:`, message, error);
  }

  warn(message: string, ...args: any[]) {
    console.warn(`${this.prefix} WARN:`, message, ...args);
  }
}
